engine/test_activation_shapes.py

```diff
@@ -0,0 +1,228 @@
+        phase = WaypointPhase.DOWN
+        curr_pos = lefmost_node_pos
+        waypoints = []
+        waypoints.append(curr_pos)
+        while (phase != WaypointPhase.TERMINATE):
+            if (phase == WaypointPhase.DOWN):
+                new_pos = (curr_pos[0]+1,curr_pos[1])
+                if curr_pos[0]+1 == rows and nodes[new_pos].activated:
+                    waypoints.append(new_pos)
+                    right_pos = return_right_position(new_pos,border_nodes)
+                    if right_pos is not None:
+                        waypoints.append(right_pos)
+                        curr_pos = right_pos
+                        phase = WaypointPhase.DOWN
+                    else:
+                        phase = WaypointPhase.TERMINATE
+                elif nodes[new_pos].activated:
+                    waypoints.append(new_pos)
+                    curr_pos = new_pos
+                else:
+                    right_pos = return_right_position(new_pos,border_nodes)
+                    if right_pos is not None:
+                        waypoints.append(right_pos)
+                        curr_pos = right_pos
+                        phase = WaypointPhase.DOWN
+                    else:
+                        phase = WaypointPhase.TERMINATE
+        return waypoints
+
+           
+
+
+
+def main():
+    rows = int(input("Num rows: "))##Good value 30
+    cols = int(input("Num cols: "))## Good value 30
+    activation_type = input("Activation type: ")
+    plt.figure()
+    plt.title("Activated Nodes")
+    plt.xlim(-1, rows+1)
+    plt.ylim(-1, cols+1)
+    plt.grid()
+    nodes = np.empty([rows, cols], dtype=object)
+    for i in range(rows):
+        for j in range(cols):
+            nodes[i,j] = Node(False,i,j)
+            
+    if activation_type == "rectangle":
+        rec_row_start = int(input("Rectangle row start: "))## Good value 10 
+        rec_row_limit = int(input("Rectangle row end: "))## Good value 20
+        rec_col_start = int(input("Rectangle col start: "))## Good value 10
+        rec_col_limit  =int(input("Rectangle col end: "))## Good value 20
+        nodes = activate_rectangle(nodes, rec_row_start, rec_col_start, rec_row_limit, rec_col_limit)
+        border_nodes,_,leftmost_node_pos, nodes = find_border_nodes(nodes)
+        way_points = get_all_lawnmower_waypoints_adjustable(nodes,leftmost_node_pos,border_nodes)
+
+    if activation_type== "circle":
+        circle_center_row = int(input("Circle center row: "))## Good value 15
+        circle_center_col = int(input("Circle center col: "))## Good value 15
+        circle_radius = int(input("Circle radius: "))## Good value 12
+        nodes = activate_circle(nodes, circle_center_row, circle_center_col, circle_radius)  
+        border_nodes,_,leftmost_node_pos, nodes = find_border_nodes(nodes)
+        way_points = get_all_lawnmower_waypoints_adjustable(nodes,leftmost_node_pos,border_nodes)
+
+    if activation_type== "triangle":
```

Comment:
lol made a straight line here w/ the triangle type, just checking if this behavior is intended to be ok?

inputs:
Num rows: 30
Num cols: 30
Activation type: triangle
x1: 1
y1: 2
x2: 15
y2: 16
x3: 29
y3: 30
<img width="752" alt="Screen Shot 2022-09-14 at 1 08 08 AM" src="https://user-images.githubusercontent.com/60713782/190064477-1abf4b66-e818-4b3d-a847-d4ffa445841f.png">

Reply:
Yeah  think this is intended